### ISSUE TITLE
Bug: Issue extending a schema

### ISSUE THREAD

**Issue by gtsueng** [biothings/discovery-app#101]
The bioschemas ComputationalTool schema will be the basis for a new outbreak schema. It has been successfully imported, registered, and visualized in the DDE; however, when you try to extend it--there is an issue with inheriting the properties from Thing. These are key properties like `name`, `url`, `identifier`, and `description` which are generally needed, but not available as seen below (Thing has 0 properties):
![cant_inherit](https://user-images.githubusercontent.com/9200138/139710238-64cb1d9a-cdf7-49e0-967b-db071867a568.jpg)

As seen in the screenshot, `Thing` has nothing to inherit. This is odd, because the Bioschemas:ComputationalTool profile includes properties like `name`, `description`, etc. that were inherited from Thing as seen below:
![image](https://user-images.githubusercontent.com/9200138/139710918-77a7d4cb-3c48-417a-bb84-3e05307591c6.png)

This is the profile I'd like to extend: https://discovery.biothings.io/view/bioschemas/ComputationalTool

**Comment by gtsueng** [comment]
Extending from the following profiles has the same issue:
- n3c:Dataset (`Thing` and `CreativeWork` completely missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140524457-9cf707d0-57d4-4007-81f5-8450a9f01741.png)
- n3c:MonetaryGrant (`Thing`, `Intangible`, `Grant` missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140529108-2caa220d-daac-4bff-82b7-5fbcc9abcc95.png)
- cvisb:Dataset (`Thing` and `CreativeWork` completely missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140525180-6fce8ef0-f7a0-4b51-8d47-c13a258ef381.png)
- cvsib:MonetaryGrant (`Thing`, `Intangible`, `Grant` missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140529475-46b178e6-2614-4c38-b3ff-0c56930eb1f9.png)
- niaid:NiaidDataset (`Thing` and `CreativeWork` completely missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140525476-787505d7-3c93-4ee9-8c25-ce02056d9ef9.png)
- niaid:Dataset (only has `CreativeWork` niaid properties and `Thing` completely missing, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140525852-ed7d6a1f-b030-44e6-a8b2-e330a83bd8f6.png)
- outbreak:ClinicalTrial (missing `MedicalEntity`, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140527583-311c4051-e785-4e95-845c-8a28b163b739.png)
- outbreak:Publication (missing `Article`, `ScholarlyArticle`, tree does not get generated)
![image](https://user-images.githubusercontent.com/9200138/140528006-06ad05da-1bf8-4060-a08d-fd4b2f3d04dc.png)
- bioschemas:BioChemEntity (`Thing` listed, but has no properties, tree DOES get generated)
![image](https://user-images.githubusercontent.com/9200138/140530665-4e84927a-810b-49b8-8e56-7a076264f790.png)
- bioschemas:Gene (`Thing` listed, but has no properties, Tree DOES get generated)
![image](https://user-images.githubusercontent.com/9200138/140530482-3f40ca81-ed76-499f-b3a9-566df46258c9.png)
- bioschemas:Protein (`Thing` listed, but no properties, Tree DOES get generated)
![image](https://user-images.githubusercontent.com/9200138/140531026-cd16aa8a-4d6f-430d-9c0e-e456d0ba73ec.png)
- google:Dataset (missing `Thing`, `CreativeWork` also no google namespace, Tree does not generate)
![image](https://user-images.githubusercontent.com/9200138/140531598-c12dd928-70e6-483f-b02a-f1db35210057.png)
- bts:Gene (all classes in tree listed, but 2 have 0 properties, Tree DOES generate)
![image](https://user-images.githubusercontent.com/9200138/140532051-35a3a2ae-962a-4289-ba9f-6e7fdced78f8.png)
- bts:Biosample (all classes in tree listed, but 2 have 0 properties, Tree DOES generate)
![image](https://user-images.githubusercontent.com/9200138/140532386-832f22a7-8731-4230-b75d-039d5f4d7ff0.png)
- **schema:Patient** (classes in tree listed, but 3 have 0 properties, Tree DOES generate)
![image](https://user-images.githubusercontent.com/9200138/140533234-0f080f26-5caa-4ab6-902e-d28ead142dc9.png)


The following profiles don't have this issue:
- ctsa:Dataset (but tree indicates I'm just extending from schemas:Dataset, not ctsa:Dataset)
![image](https://user-images.githubusercontent.com/9200138/140524914-0f88e64d-124d-413f-afa3-3b3ff0acf8b9.png)
- outbreak:Dataset (`CreativeWork` only has 1 property [the one included in outbreak] but otherwise it seems to be working)
![image](https://user-images.githubusercontent.com/9200138/140527030-aad17ae3-4340-4610-a97f-58a5988f1f09.png)
- outbreak:Analysis (`CreativeWork` only has 1 property [the one included in outbreak] but otherwise it seems to be working)
- outbreak:Protocol (`CreativeWork` only has 1 property [the one included in outbreak] but otherwise it seems to be working)
- cvisb-patient:Patient

**Comment by gtsueng** [comment]
One thing to note is that all schema.org classes were initially thought to not have this issue. However, it appears that heavily nested schema.org classes may also have this issue (eg- schema:Patient).

**Comment by nsjuty** [comment]
Hi all - just wondering if there is an update on this bug? I'm blocked on a few tasks on Bioschemas extensions at the moment...
thanks!
Nick

**Comment by marcodarko** [comment]
Thank you for your patience, issue will be fixed soon I have opened a PR will tag you when it's ready!  
<img width="801" alt="Screen Shot 2021-11-17 at 11 48 47 AM" src="https://user-images.githubusercontent.com/23092057/142272284-55fb61ae-97a9-48f8-b433-9efe4eeb45f5.png">

**Comment by marcodarko** [comment]
@gtsueng @nsjuty Update is live now and it appears to be fixed, please check it out and feel free to close once you confirm this addresses this issue.

**Comment by gtsueng** [comment]
@marcodarko The inherited properties are now viewable and can be extended via the DDE; however, the resulting extended json is not functional as the inherited properties in the validation are not properly referencing their source.

For example, I created [this schema](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/niaidCompToolsTest.json) within the DDE by extending the bioschemas:ComputationalTool profile
But, as you can see, even the most basic field "name" which is described as:
```      "$validation": {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "name": {
            "description": "Descriptive name of the computational tool",
            "type": "string"
          },
```
Results in the following error:
![image](https://user-images.githubusercontent.com/9200138/142478222-ad231450-97ec-419d-8dde-f74eee19d0a8.png)

For trouble-shooting purposes, here is a simpler test schema that was created by extending the bioschemas:ComputationalTool schema. https://github.com/gtsueng/niaid-data-portal/blob/master/DDE_compToolTest.jsonld
For this one, a single property was selected from each class in the hierarchy.

**Comment by marcodarko** [comment]
I guess there's already an issue for this: https://github.com/biothings/discovery-app/issues/77 so we can close this if the particular issue is resolved.

**Comment by gtsueng** [comment]
From what I've seen of the issue, it looks like all properties in the validation are affected, and the DDE will whine about the first property on the list, and move downwards if a problematic property is somehow addressed.

For example, [I rearranged the order of the properties](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTest2.json) in the $validation of so that `name` is first, and the DDE will whine about `name` instead of `codeRepository`

Some info that might help diagnose the issue:
The Bioschemas Gene class (https://github.com/BioSchemas/specifications/blob/master/Gene/jsonld/Gene_v1.0-RELEASE.json) exhibits this error in the DDE; however, this class is automatically copied into the bioschemas json schema and successfully displays in the DDE (https://github.com/BioSchemas/bioschemas-dde/blob/main/bioschemas.json)

This suggests that the issue is at least partly due to broken referencing. To test this, I applied some potential fixes for our generated schema that has this issue: https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTest.jsonld

These are the potential schema fixes and the results of the test:
1. [Add the missing property `codeRepository` into the graph as a property to be defined in the schema](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix1.json). Result-- error persists
2.  [Add a dummy version of the parent class in hopes it references properly](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix2.json). Result-- error persists
3.  [Add a dummy version of the parent class in hopes it references properly AND add the missing property `codeRepository` into the graph as a property to be defined in the schema](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix3.json). Result-- Fixes error
4.  [Add nothing but a reference to the bioschemas class in the `@context`](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix4.json). Result-- error persists
5. [Add reference to the bioschemas class in the `@context` and as a `$bioschemas`](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix5.json). Result--error persists
6. [Add reference to the bioschemas class in the `@context` and add the missing property `codeRepository` into the graph.](https://github.com/gtsueng/niaid-data-portal/blob/master/schema/DDE_compToolTestFix6.json) Result-- error persists

From this test it looks like, both the non-schema.org parent class (or at least a dummy version of it) and any non-schema.org properties inherited from that class need to be included in order for it to reference and validate properly. Notice that the addition of the dummy `bioschemas:ComputationalTool` class and the `codeRepository` property specific to that class had to be added in order to bypass this issue, but the other properties inherited from schema.org into the bioschemas class no longer raise an issue in this case. Previously, all properties were problematic, even `name`.

**Comment by gtsueng** [comment]
closing this issue as the related bug (https://github.com/biothings/discovery-app/issues/77) has its own ticket
